### PR TITLE
fix undef showing

### PR DESCRIPTION
**Pull request by Moelf**
main issue was that error happens at assign:
https://github.com/fonsp/Pluto.jl/blob/b53d282a132752e34563593b8c7c294866042efb/src/runner/PlutoRunner.jl#L372

so instead of broadcast over `zip` of (index, elements), a specialized function for array is implemented.

Also, I think a lot can be re-used from `Base`, for example:
```
julia> Base.print_matrix(stdout, vvv)
 #undef
 #undef
 #undef
```
Maybe for the next re-factor :construction_worker:

Also I switched to `@view` when possible, since users can have huge array (maybe each element is big or something), so slicing with copy was not as good of an idea (I think).

After this patch:
image

### CODE REVIEW

**Comment by Moelf** [comment]
fix #352

**Comment by fonsp** [comment]
Nice! Could you add a testset?

https://github.com/fonsp/Pluto.jl/blob/master/test/RichOutput.jl

**Comment by Moelf** [comment]
@fonsp done

File: src/runner/PlutoRunner.jl [hunk 1]
```diff
@@ -375,6 +375,19 @@ function show_array_row(io::IO, pair::Tuple)
     print(io, "</v></r>")
 end
 
+function show_array_row(io::IO, idxs::AbstractVector{<:Integer}, ary::AbstractArray{<:Any, 1})
```
Comment (fonsp) [inline src/runner/PlutoRunner.jl:378]: This should be a new function: it does not show a `row`.

File: src/runner/PlutoRunner.jl [hunk 2]
```diff
@@ -375,6 +375,19 @@ function show_array_row(io::IO, pair::Tuple)
...
+    for i in idxs
+        if isassigned(ary, i)
+            show_array_row(io, (i, ary[i]))
+        else
+            # such that #undef is not confused with "#undef"
+            print(io, "<r><k>", i, "</k><v>")
+            print(io, Base.undef_ref_str)
+            print(io, "</v></r>")
```
Comment (fonsp) [inline src/runner/PlutoRunner.jl:386]:
```julia
show_array_row(io, (i, Text(Base.undef_ref_str)))
```

**Review by fonsp — CHANGES_REQUESTED** [review]

File: src/runner/PlutoRunner.jl (hunk 1 above)
Comment (Moelf) [reply src/runner/PlutoRunner.jl:378]: I can rename it to "rows" but that's too subtle of a difference?

**Comment by fonsp** [comment]
Can you remove abbreviations? idx, ary, els

**Comment by Moelf** [comment]
> Can you remove abbreviations? idx, ary, els

yes, where and why? I mean we used `i, el = pair` before without a problem right?

**Comment by fonsp** [comment]
Awesome, thanks!

**Pull request merged**